partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt

```diff
@@ -817,7 +819,16 @@ internal class PlanTyper(
         }
 
         override fun visitRexOpPivot(node: Rex.Op.Pivot, ctx: StaticType?): Rex {
-            TODO("Type RexOpPivot")
+            val rel = node.rel.type(locals)
+            val typeEnv = TypeEnv(rel.type.schema, ResolutionStrategy.LOCAL)
+            val key = node.key.type(typeEnv)
+            val value = node.value.type(typeEnv)
+            val type = StructType(
+                contentClosed = false,
+                constraints = setOf(TupleConstraint.Open(true))
```

Comment:
Should a `fields` value be added for the key-value pair? SchemaInference test for `pivot-00` currently doesn't include the field for `t.a`: `t.c`.

Reply:
We don't know the fields of a PIVOT until we evaluate the query